app/src/main/java/com/example/materialcomposereader/data/Resource.kt

```diff
@@ -0,0 +1,7 @@
+package com.example.materialcomposereader.data
+
+sealed class Resource<T>(val data: T? = null, val message: String? = null) {
+    class Success<T>(data: T) : Resource<T>(data)
+    class Error<T>(message: String?, data: T? = null) : Resource<T>(data, message)
+    class Loading<T>(data: T? = null) : Resource<T>(data)
+}
```

Comment:
It's still possible to create a `Resource.Error(message = "My message", data = "Some data")` and a `Resource.Loading("data"). We want to define errors out of existence using our type system. 

What about something like:
```suggestion
sealed class Resource {
    data class Success<T>(val data: T) : Resource()
    data class Error(val message: String?) : Resource()
    object Loading : Resource()
}
```

Then you get:

```kotlin
val success = Resource.Success("hello") // the "argument here must be some data
val error = Resource.Error("Bad error") // the argument here must be a message
val loading = Resource.Loading
```

Reply:
On second thought, it might need to be something like:

```kotlin
sealed class Resource<T> {
    data class Success<T>(val data: T) : Resource<T>()
    data class Error<T>(val message: String?) : Resource<T>()
    class Loading<T> : Resource<T>()
}
```
so that you can retain the type info. I suggest you merge for now and then try using it in your next PR so you can fix it up if necessary